### PR TITLE
Have crew monitor display entity coordinates instead

### LINKED ISSUE #13042: Crew monitoring should be using entitycoordinates
Right now it uses mapcoordinates but it really shouldn't be.

Also it displays world positions which is unusable ingame.

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR 

closes #13042 

Crew monitors now display entity coordinates instead of map coordinates

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:

### CODE REVIEW

**Comment by moonheart08** [comment]
Have you tested what this shows for someone inside a locker?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
You should remove mapcoordinates entirely and only send entitycoordinates. It should be sending entitycoordinates relative to grid / map as well to avoid pvs issues (use GetMoverCoordinates on transformsystem).

**Review by DrSmugleaf — APPROVED** [review]

File: Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs [hunk 1]
```diff
@@ -90,14 +92,15 @@ private BoxContainer GetPositionBox(MapCoordinates? coordinates, Vector2 sensorP
...
+                var mapCoords = coordinates.Value.ToMap(_entityManager).Position;
                 var dirIcon = new DirectionIcon(snap, precision)
                 {
                     SetSize = (IconSize, IconSize),
                     Margin = new(0, 0, 4, 0)
                 };
                 box.AddChild(dirIcon);
                 box.AddChild(new Label() { Text = pos.ToString() });
-                _directionIcons.Add((dirIcon, coordinates.Value.Position - sensorPosition));
+                _directionIcons.Add((dirIcon, mapCoords - sensorPosition));
```
Comment (metalgearsloth) [inline Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:103]: Is this correct? Just doing mapcoords - mapcoords won't give the correct rotation in your frame of reference.

File: Content.Server/Medical/CrewMonitoring/CrewMonitoringConsoleSystem.cs [hunk 2]
```diff
@@ -66,9 +67,9 @@ private void UpdateUserInterface(EntityUid uid, CrewMonitoringConsoleComponent?
                 return;
 
             // update all sensors info
-            var worldPos = _xform.GetWorldPosition(uid);
+            var xform = Transform(uid);
             var allSensors = component.ConnectedSensors.Values.ToList();
-            var uiState = new CrewMonitoringState(allSensors, worldPos, component.Snap, component.Precision);
+            var uiState = new CrewMonitoringState(allSensors, xform.WorldPosition, component.Snap, component.Precision);
```
Comment (metalgearsloth) [inline Content.Server/Medical/CrewMonitoring/CrewMonitoringConsoleSystem.cs:72]: Needs updating?

File: Content.Server/Medical/SuitSensors/SuitSensorSystem.cs [hunk 3]
```diff
@@ -228,7 +228,7 @@ private string GetModeName(SuitSensorMode mode)
                 return null;
 
             // check if sensor is enabled and worn by user
-            if (sensor.Mode == SuitSensorMode.SensorOff || sensor.User == null)
+            if (sensor.Mode == SuitSensorMode.SensorOff || sensor.User == null || transform.GridUid == null)
```
Comment (metalgearsloth) [inline Content.Server/Medical/SuitSensors/SuitSensorSystem.cs:231]: Don't sensors still work in space in ss13?

File: Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs (hunk 1 above)
Comment (keronshb) [reply Content.Client/Medical/CrewMonitoring/CrewMonitoringWindow.xaml.cs:103]: Seems like it's correct. The other way wasn't giving me the correct position.

File: Content.Server/Medical/CrewMonitoring/CrewMonitoringConsoleSystem.cs (hunk 2 above)
Comment (keronshb) [reply Content.Server/Medical/CrewMonitoring/CrewMonitoringConsoleSystem.cs:72]: What do you mean? It works fine in game.

File: Content.Server/Medical/SuitSensors/SuitSensorSystem.cs (hunk 3 above)
Comment (keronshb) [reply Content.Server/Medical/SuitSensors/SuitSensorSystem.cs:231]: IIRC this threw some sort of error if I didn't check for a null grid.

**Pull request merged**